optax/losses/_segmentation.py

```diff
@@ -0,0 +1,244 @@
+
+    Args:
+      predictions: Logits of shape [..., num_classes] for multi-class or
+                  [..., 1] or [...] for binary segmentation.
+      targets: One-hot encoded targets of shape [..., num_classes] for
+              multi-class or binary targets of shape [..., 1] or [...] for
+              binary.
+      class_weights: Optional weights for each class of shape [num_classes].
+                    If None, all classes weighted equally.
+      smooth: Smoothing parameter to avoid division by zero and improve
+             gradient stability.
+      apply_softmax: Whether to apply softmax to predictions. Set False if
+                    predictions are already probabilities.
+      reduction: How to reduce across classes: 'mean', 'sum', or 'none'.
+                'none' returns per-class losses.
+      ignore_background: If True, excludes the first class (index 0) from loss
+                        computation. Useful when class 0 represents background.
+
+    Returns:
+      Loss values. Shape depends on reduction:
+
+      - 'mean'/'sum': [...] (batch dimensions only)
+      - 'none': [..., num_classes] (includes class dimension)
+
+    Examples:
+      Binary segmentation:
+
+      >>> logits = jnp.array([[1.0, -1.0], [0.5, 0.5]])  # Shape: [2, 2]
+      >>> targets = jnp.array([[1.0, 0.0], [1.0, 0.0]])  # Shape: [2, 2]
+      >>> loss = dice_loss(logits[..., None], targets[..., None])
+
+      Multi-class segmentation:
+
+      >>> key = jax.random.PRNGKey(0)
+      >>> logits = jax.random.normal(key, (4, 64, 64, 3))  # 4 samples, 3 cls
+      >>> targets = jax.nn.one_hot(labels, 3)  # One-hot encoded
+      >>> loss = dice_loss(logits, targets)
+
+    References:
+      Milletari et al. "V-Net: Fully Convolutional Neural Networks for
+      Volumetric Medical Image Segmentation" (2016).
+    """
+
+    if predictions.ndim == targets.ndim - 1:
+        predictions = predictions[..., None]
+    if targets.ndim == predictions.ndim - 1:
+        targets = targets[..., None]
+
+    chex.assert_equal_shape([predictions, targets])
+
+    # Input validation for probability distributions
+    if not apply_softmax:
+        # Tolerance for numerical stability
+        pred_sums = jnp.sum(predictions, axis=-1)
+        if not jnp.allclose(pred_sums, 1.0, rtol=1e-5):
+            raise ValueError(
+                "When apply_softmax=False, predictions must be valid "
+                "probability distributions that sum to 1 along the class axis. "
+                f"Found sum range: [{jnp.min(pred_sums):.6f}, "
+                f"{jnp.max(pred_sums):.6f}]"
+            )
+
+    # Convert logits to probabilities
+    if apply_softmax:
+        if predictions.shape[-1] == 1:
+            # Binary case - use sigmoid
+            probs = jax.nn.sigmoid(predictions)
+        else:
+            # Multi-class case - use softmax
+            probs = jax.nn.softmax(predictions, axis=-1)
+    else:
+        probs = predictions
```

Comment:
Use `jax.lax.cond` to remove these conditional branches to pass the `pylint` tests.
```suggestion
    probs = jax.lax.cond(
        apply_softmax,
        lambda preds: jax.lax.cond(
            preds.shape[-1] == 1,
            jax.nn.sigmoid,
            lambda p: jax.nn.softmax(p, axis=-1),
            preds
        ),
        lambda preds: preds,
        predictions
    )
```